pkg/exec/builder/execute.go

```diff
@@ -77,6 +78,11 @@ func ExecuteStep(cxt *context.Context, step ExecutableStep) (string, error) {
 	}
 	args = append(args, flags.ToSlice(dashes)...)
 
+	// Split up any arguments or flags that have spaces so that we pass them as separate array elements
+	// It doesn't show up any differently in the printed command, but it matters to how the command
+	// it executed against the system.
+	args = expandOnWhitespace(args)
```

Comment:
Do you think passing the command off to the shell (which parses this for us) be something that would work here instead of parsing the command ourselves? Looks similar to the problem I ran into here: https://github.com/deislabs/porter/pull/921/files#diff-9cde54dd12e8d18ebea48c996aa54c92R45

Reply:
@halkyon Do you have a cross platform way of doing that? I was looking for ideas yesterday on how to make the shell deal with this instead of that kludge function (that doesn't work anyway). I would love some help fixing this problem! 😂

Reply:
It is so tempting to be like `bash -c "dump all their stuff here"` but actually their command may also be a bash -c "..." command too. Or it could be calling to a binary that is in a scratch based invocation image.

Reply:
Ah, right, I didn’t realise the environments commands could be running in, still getting my head around porter. Sorry for butting in a bit on this PR by the way!

Reply:
Oh hey no worries, I liked the suggestion and would have used it in a heartbeat if I could!